### ISSUE TITLE
Problem using defprocessor

### ISSUE THREAD

**Issue by escherize** [nathell/skyscraper#1]
Greetings,

Appreciate your library, the idea behind it is swell.  I am getting an error when using defprocessor, I've checked multiple times and always seem to get an error when using anything besides the forest example.

Running:

``` clojure
(defprocessor hn-index
  :cache-template "hn/index"
  :process-fn (fn [res context]
                (println res)
                (println context)
                {:foo"bar"}))

(scrape [{:url "https://news.ycombinator.com", :processor :hn-index}])

```

Gives me this output:

```

1. Unhandled clojure.lang.ArityException
   Wrong number of args (1) passed to: etsy/hn-index/fn--22111

                      AFn.java:  429  clojure.lang.AFn/throwArity
                      AFn.java:   32  clojure.lang.AFn/invoke
                skyscraper.clj:  126  skyscraper/processor
                   RestFn.java:  521  clojure.lang.RestFn/invoke
                          REPL:    1  scrape.skyscraper.etsy/hn-index
                      Var.java:  383  clojure.lang.Var/invoke
                skyscraper.clj:  144  skyscraper/do-scrape/fn
                      core.clj: 2622  clojure.core/map/fn
                  LazySeq.java:   40  clojure.lang.LazySeq/sval
                  LazySeq.java:   49  clojure.lang.LazySeq/seq
                       RT.java:  507  clojure.lang.RT/seq
                      core.clj:  137  clojure.core/seq
                      core.clj:  630  clojure.core/apply
                      core.clj: 2660  clojure.core/mapcat
                   RestFn.java:  423  clojure.lang.RestFn/invoke
                skyscraper.clj:  140  skyscraper/do-scrape
                skyscraper.clj:  152  skyscraper/scrape
                   RestFn.java:  410  clojure.lang.RestFn/invoke
                          REPL:    1  scrape.skyscraper.etsy/eval22114
                 Compiler.java: 6782  clojure.lang.Compiler/eval
                 Compiler.java: 6745  clojure.lang.Compiler/eval
                      core.clj: 3081  clojure.core/eval
                      main.clj:  240  clojure.main/repl/read-eval-print/fn
                      main.clj:  240  clojure.main/repl/read-eval-print
                      main.clj:  258  clojure.main/repl/fn
                      main.clj:  258  clojure.main/repl
                   RestFn.java: 1523  clojure.lang.RestFn/invoke
        interruptible_eval.clj:   58  clojure.tools.nrepl.middleware.interruptible-eval/evaluate/fn
                      AFn.java:  152  clojure.lang.AFn/applyToHelper
                      AFn.java:  144  clojure.lang.AFn/applyTo
                      core.clj:  630  clojure.core/apply
                      core.clj: 1868  clojure.core/with-bindings*
                   RestFn.java:  425  clojure.lang.RestFn/invoke
        interruptible_eval.clj:   56  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
        interruptible_eval.clj:  191  clojure.tools.nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
        interruptible_eval.clj:  159  clojure.tools.nrepl.middleware.interruptible-eval/run-next/fn
                      AFn.java:   22  clojure.lang.AFn/run
       ThreadPoolExecutor.java: 1142  java.util.concurrent.ThreadPoolExecutor/runWorker
       ThreadPoolExecutor.java:  617  java.util.concurrent.ThreadPoolExecutor$Worker/run
                   Thread.java:  745  java.lang.Thread/run


1. Unhandled clojure.lang.ArityException
   Wrong number of args (1) passed to: etsy/hn-index/fn--22111

                      AFn.java:  429  clojure.lang.AFn/throwArity
                      AFn.java:   32  clojure.lang.AFn/invoke
                skyscraper.clj:  126  skyscraper/processor
                   RestFn.java:  521  clojure.lang.RestFn/invoke
                          REPL:    1  scrape.skyscraper.etsy/hn-index
                      Var.java:  383  clojure.lang.Var/invoke
                skyscraper.clj:  144  skyscraper/do-scrape/fn
                      core.clj: 2622  clojure.core/map/fn
                  LazySeq.java:   40  clojure.lang.LazySeq/sval
                  LazySeq.java:   49  clojure.lang.LazySeq/seq
                       RT.java:  507  clojure.lang.RT/seq
                      core.clj:  137  clojure.core/seq
                      core.clj:  630  clojure.core/apply
                      core.clj: 2660  clojure.core/mapcat
                   RestFn.java:  423  clojure.lang.RestFn/invoke
                skyscraper.clj:  140  skyscraper/do-scrape
                skyscraper.clj:  152  skyscraper/scrape
                   RestFn.java:  410  clojure.lang.RestFn/invoke
                          REPL:    1  scrape.skyscraper.etsy/eval22114
                 Compiler.java: 6782  clojure.lang.Compiler/eval
                 Compiler.java: 6745  clojure.lang.Compiler/eval
                      core.clj: 3081  clojure.core/eval
                      main.clj:  240  clojure.main/repl/read-eval-print/fn
                      main.clj:  240  clojure.main/repl/read-eval-print
                      main.clj:  258  clojure.main/repl/fn
                      main.clj:  258  clojure.main/repl
                   RestFn.java: 1523  clojure.lang.RestFn/invoke
        interruptible_eval.clj:   58  clojure.tools.nrepl.middleware.interruptible-eval/evaluate/fn
                      AFn.java:  152  clojure.lang.AFn/applyToHelper
                      AFn.java:  144  clojure.lang.AFn/applyTo
                      core.clj:  630  clojure.core/apply
                      core.clj: 1868  clojure.core/with-bindings*
                   RestFn.java:  425  clojure.lang.RestFn/invoke
        interruptible_eval.clj:   56  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
        interruptible_eval.clj:  191  clojure.tools.nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
        interruptible_eval.clj:  159  clojure.tools.nrepl.middleware.interruptible-eval/run-next/fn
                      AFn.java:   22  clojure.lang.AFn/run
       ThreadPoolExecutor.java: 1142  java.util.concurrent.ThreadPoolExecutor/runWorker
       ThreadPoolExecutor.java:  617  java.util.concurrent.ThreadPoolExecutor$Worker/run
                   Thread.java:  745  java.lang.Thread/run
```

I had a look at processor, but I can't figure out what is causing this issue.

**Comment by nathell** [comment]
Hi Bryan,

First off, thanks for your interest in Skyscraper. It's terrific to know someone is actually trying out the code!

The issue you've encountered is an unfortunate side-effect of the fact that Skyscraper has had some API-breaking commits since 0.1.0 (the only release so far) -- you're attempting to use the 0.1.0 release with the current docs. In 0.1.0, there was no way to access the current context from within a processor; it turns out it's useful in certain use cases (I have since encountered such a case).

This is the relevant commit: https://github.com/nathell/skyscraper/commit/0fd1392806442366922266b4b5a425f0c6b7a2b4. As there were some other fixes, I guess it's high time to release 0.1.1, which I'll do very shortly.

**Comment by nathell** [comment]
I've just released 0.1.1, which contains the goodies I'd mentioned.
